MFVNeutralino/plugins/VertexAuxProducer.cc

```diff
@@ -196,32 +262,68 @@ void MFVVertexAuxProducer::produce(edm::Event& event, const edm::EventSetup& set
+        //need to double check track ids w/ types
+        reco::TrackRef tk = it->castTo<reco::TrackRef>();
+
+        // using generalized function to separate into track types
+        if (track_rescaler_which == 1){
+          if ((tk.id().id() == 155) & (tk->pt() >= 20.0)) {
+            rs_ttks.push_back(tt_builder->build(track_rescaler.scale(**it, "electron").rescaled_tk));
+          }
+          else if ((tk.id().id() == 156) & (tk->pt() >= 20.0))  {
+            rs_ttks.push_back(tt_builder->build(track_rescaler.scale(**it, "muon").rescaled_tk));
+          }
+          else {
+            rs_ttks.push_back(tt_builder->build(track_rescaler.scale(**it, "").rescaled_tk));
+          }
+        }
+        else rs_ttks.push_back(tt_builder->build(track_rescaler.scale(**it).rescaled_tk));
+
       }
+      //get seed tracks outside all vertices
+      size_t sedtki = 0;
+      for (const reco::Track& sedtk : *vertex_seed_tracks) {
+	      assert(abs(sedtk.charge()) == 1);
+	      if ((fabs(sedtk.pt() - fabs(tk->charge() * tk->pt())) < 0.0001 &&
+		    fabs(sedtk.eta() - tk->eta()) < 0.0001 &&
+		    fabs(sedtk.phi() - tk->phi()) < 0.0001) || std::count(vec_outsedtki.begin(), vec_outsedtki.end(), sedtki) > 0) {
+		    continue;
+	      }
+	      vec_outsedtki.push_back(sedtki);
+	      sedtki++;
+      }
+    }
     if (rs_ttks.size() > 1) {
-      reco::Vertex rs_sv(TransientVertex(kv_reco->vertex(rs_ttks)));
-      if (rs_sv.isValid()) {
-        const auto d = distcalc_2d.distance(rs_sv, fake_bs_vtx);
-        aux.rescale_chi2 = rs_sv.chi2();
-        aux.rescale_x = rs_sv.x();
-        aux.rescale_y = rs_sv.y();
-        aux.rescale_z = rs_sv.z();
-        aux.rescale_cxx = rs_sv.covariance(0,0);
-        aux.rescale_cxy = rs_sv.covariance(0,1);
-        aux.rescale_cxz = rs_sv.covariance(0,2);
-        aux.rescale_cyy = rs_sv.covariance(1,1);
-        aux.rescale_cyz = rs_sv.covariance(1,2);
-        aux.rescale_czz = rs_sv.covariance(2,2);
-        aux.rescale_bs2ddist = d.value();
-        aux.rescale_bs2derr = d.error();
+      try {
+        reco::Vertex rs_sv(TransientVertex(kv_reco->vertex(rs_ttks)));
+        if (rs_sv.isValid()) {
+          const auto d = distcalc_2d.distance(rs_sv, fake_bs_vtx);
+          aux.rescale_chi2 = rs_sv.chi2();
+          aux.rescale_x = rs_sv.x();
+          aux.rescale_y = rs_sv.y();
+          aux.rescale_z = rs_sv.z();
+          aux.rescale_cxx = rs_sv.covariance(0,0);
+          aux.rescale_cxy = rs_sv.covariance(0,1);
+          aux.rescale_cxz = rs_sv.covariance(0,2);
+          aux.rescale_cyy = rs_sv.covariance(1,1);
+          aux.rescale_cyz = rs_sv.covariance(1,2);
+          aux.rescale_czz = rs_sv.covariance(2,2);
+          aux.rescale_bs2ddist = d.value();
+          aux.rescale_bs2derr = d.error();
+        }
+        else {
+          aux.rescale_chi2 = -1;
+        }
       }
-      else
-        aux.rescale_chi2 = -1;
+      catch (...) {}
```

Comment:
This try catch block look a little scary, since it will silently fail if we have an issue. Do we actually need it? Not sure if this is one for @abbywarden @pkotamnives or @duquale